packages/contracts-bedrock/src/L1/StandardValidator.sol

```diff
@@ -0,0 +1,271 @@
+        internalRequire(_portal.guardian() == superchainConfig.guardian(), "PORTAL-60");
+        internalRequire(_portal.paused() == superchainConfig.paused(), "PORTAL-70");
+        internalRequire(_portal.l2Sender() == defaultL2Sender, "PORTAL-80");
+    }
+
+    function assertValidDisputeGameFactory(ISystemConfig _sysCfg, address _l1PAO) internal {
+        IDisputeGameFactory _factory = IDisputeGameFactory(_sysCfg.disputeGameFactory());
+        internalRequire(stringEq(_factory.version(), disputeGameFactoryVersion), "DF-10");
+        internalRequire(_factory.owner() == address(_l1PAO), "DF-20");
+    }
+
+    function assertValidPermissionedDisputeGame(
+        ISystemConfig _sysCfg,
+        address _l1PAO,
+        bytes32 _absolutePrestate,
+        uint256 _l2ChainID
+    )
+        internal
+    {
+        IDisputeGameFactory _factory = IDisputeGameFactory(_sysCfg.disputeGameFactory());
+        IPermissionedDisputeGame _game =
+            IPermissionedDisputeGame(address(_factory.gameImpls(GameTypes.PERMISSIONED_CANNON)));
+
+        if (address(_game) == address(0)) {
+            internalRequire(false, "PDG-10");
+            // Return early to avoid reverting, since this means that there is no valid game impl
+            // for this game type.
+            return;
+        }
+
+        bool validGameVM = address(_game.vm()) == address(mips);
+
+        internalRequire(stringEq(_game.version(), permissionedDisputeGameVersion), "PDG-20");
+        internalRequire(bytecodeHashEq(address(_game), permissionedDisputeGameHash), "PDG-30");
+        IAnchorStateRegistry _asr = _game.anchorStateRegistry();
+        internalRequire(GameType.unwrap(_game.gameType()) == GameType.unwrap(GameTypes.PERMISSIONED_CANNON), "PDG-40");
+        internalRequire(Claim.unwrap(_game.absolutePrestate()) == _absolutePrestate, "PDG-50");
+        internalRequire(validGameVM, "PDG-60");
+        internalRequire(_game.l2ChainId() == _l2ChainID, "PDG-70");
+        internalRequire(_game.l2BlockNumber() == 0, "PDG-80");
+        internalRequire(Duration.unwrap(_game.clockExtension()) == 10800, "PDG-90");
+        internalRequire(_game.splitDepth() == 30, "PDG-100");
+        internalRequire(_game.maxGameDepth() == 73, "PDG-110");
+        internalRequire(Duration.unwrap(_game.maxClockDuration()) == 302400, "PDG-120");
+
+        assertValidDelayedWETH(_game.weth(), _l1PAO);
+        assertValidAnchorStateRegistry(_factory, _asr);
+
+        // Only assert valid preimage oracle if the game VM is valid, since otherwise
+        // the contract is likely to revert.
+        if (validGameVM) {
+            assertValidPreimageOracle(_game.vm().oracle());
+        }
+    }
+
+    function assertValidDelayedWETH(IDelayedWETH _weth, address _l1PAO) internal {
+        internalRequire(stringEq(_weth.version(), delayedWETHVersion), "DWETH-10");
+        internalRequire(_weth.owner() == _l1PAO, "DWETH-20");
```

Comment:
We also need to check that the `delay()` is `1 weeks`

Reply:
done